docs/agent-http-api.md

```diff
@@ -0,0 +1,94 @@
+# Agent HTTP API
+
+This page documents the structure of the HTTP API used by agents to communicate
+with the crater server.
+
+The base URL for the Agent API is `/agent-api/`.
+
+## Authentication
+
+All the requests to the Agent API are restricted to authenticated clients only.
+Each agent should have its own unique API token. To authenticate you need to
+provide the `Authorization` HTTP header with the right token:
+
+```
+Authorization: token YOUR-AGENT-TOKEN
```

Comment:
I don't mind us using a custom `Authorization` scheme, but can we namespace it a bit more? E.g. `CraterToken`.

Reply:
Yeah, fixed.